src/librustc_mir/interpret/validity.rs

```diff
@@ -140,6 +141,33 @@ fn scalar_format<Tag>(value: ScalarMaybeUndef<Tag>) -> String {
     }
 }
 
+fn wrapping_range_contains(r: &RangeInclusive<u128>, test: u128) -> bool {
+    let (lo, hi) = r.clone().into_inner();
+    if lo > hi {
+        // Wrapped
+        (..=hi).contains(&test) || (lo..).contains(&test)
+    } else {
+        // Normal
+        r.contains(&test)
+    }
+}
+
+// Formats such that a sentence like "expected something {}" to mean
+// "expected something <in the given range>" makes sense.
+fn wrapping_range_format(r: &RangeInclusive<u128>, max_hi: u128) -> String {
+    let (lo, hi) = r.clone().into_inner();
+    debug_assert!(hi <= max_hi);
+    if lo > hi {
+        format!("less or equal to {}, or greater or equal to {}", hi, lo)
+    } else {
+        if hi == max_hi {
+            format!("greater or equal to {}", lo)
+        } else {
+            format!("in the range {:?}", r)
```

Comment:
all the `IndexVec` indices are `smaller or equal to`. Seems like something reasonable to support, too?

Reply:
I can add the printing, but how would I go about testing it?

Reply:
Oh actually we have tons of tests for this, from `NonZero`.